src/appleseed/renderer/modeling/project/projectfileupdater.cpp

```diff
@@ -1786,6 +1786,90 @@ namespace
             params.remove_path("render_stamp_format");
         }
     };
+
+    //
+    // Update from revision 26 to revision 27.
+    //
+
+    class UpdateFromRevision_26
+      : public Updater
+    {
+      public:
+        explicit UpdateFromRevision_26(Project& project)
+          : Updater(project, 26)
+        {
+        }
+
+        void update() override
+        {
+            update_diagnostic_option();
+            update_pixel_renderer_option();
+        }
+
+      private:
+
+        // Merge frame::save_extra_aovs and PixelRenderer::enable_diagnostics.
+        void update_diagnostic_option()
+        {
+            for (each<ConfigurationContainer> i = m_project.configurations(); i; ++i)
+            {
+                Dictionary& root = i->get_parameters();
+
+                if (root.dictionaries().exist("uniform_pixel_renderer"))
+                {
+                    Dictionary& gtr = root.dictionary("uniform_pixel_renderer");
+
+                    if (gtr.strings().exist("enable_diagnostics"))
+                        gtr.strings().remove("enable_diagnostics");
+                }
+
+                if (root.dictionaries().exist("adaptive_pixel_renderer"))
+                {
+                    Dictionary& gtr = root.dictionary("adaptive_pixel_renderer");
+
+                    if (gtr.strings().exist("enable_diagnostics"))
+                        gtr.strings().remove("enable_diagnostics");
+                }
+            }
+
+            Frame* frame = m_project.get_frame();
+
+            if (frame == nullptr)
+                return;
+
+            ParamArray& frame_params = frame->get_parameters();
+
+            if (frame_params.strings().exist("save_extra_aovs"))
+            {
+                // The frame does not reference any camera: use the first camera.
+                frame_params.insert(
+                    "enable_diagnostic_aovs",
+                    frame_params.get<bool>("save_extra_aovs"));
+
+                frame_params.strings().remove("save_extra_aovs");
+            }
+        }
+
+        // Rename pixel_renderer to sampling_method.
```

Comment:
Because if we do `adaptive_tile` we don't use any `PixelRenderer`.

Reply:
> Because if we do adaptive_tile we don't use any PixelRenderer.

I don't understand how that's an argument to rename `pixel_renderer` to `sampling_method`. In all other cases we do have pixel renderers.

Let's revert this.

Reply:
I find this weird to name it pixel_renderer because if `pixel_renderer = adaptive_tile` we don't use any pixel renderer..

Reply:
I don't follow. AdaptiveTileRenderer is a tile renderer so it's the `tile_renderer` parameter that must be set to `adaptive_tile_renderer` and `pixel_renderer` must not be set at all.

Reply:
The hierarchy of renderer components (frame/tile/pixel/sample) described in https://github.com/appleseedhq/appleseed/wiki/Renderer-Components is fixed. You simply plug yourself at the tile level instead of at the pixel level.

Reply:
If you imagine the user behavior maybe you will see my point of view. When you are in the ui, chossing for the renderer, you have only one dropdown that switches between `uniform`, `adaptive` and `adaptive_tile`. For me this is the method and depending on the method, we choose correct pixel and tile renderers

Reply:
From a user's point of view, the hierarchy of frame/tile/pixel/sample renderers is clearly irrelevant. What we do under the hood is separate from what how we present things to the user.

Reply:
So the problem is in the UI. I should not link the dropdown to `pixel_renderer` but instead do a regular dropdown and edit `pixel_renderer` or `tile_renderer` accordingly.

Reply:
Exactly.